### PR TITLE
added imm service

### PR DESCRIPTION
**Pull request by twurman**
Added the imm service. Includes a common folder which both sirius/ and template/ files use. template/ has been tested and is a standalone service with a client included for demonstration purposes. sirius/ contains the files necessary to run the sirius imm service. read the READMEs for compilation and run instructions

### CODE REVIEW

File: imm/sirius/Makefile [hunk 1]
```diff
@@ -0,0 +1,64 @@
+CXX = g++
+PROTO = protoc
+CXX_FLAGS = -I/usr/local/boost_1_58_0
```
Comment (jhauswald) [inline imm/sirius/Makefile]: make 1 boost location and use it across makefile

File: imm/sirius/service.thrift [hunk 2]
```diff
@@ -0,0 +1,9 @@
+#
+#
+#
+# The ImageMatchingService is running on the server to accept image queries
+# and return the name that best matches in the pre-trained database  
+service ImageMatchingService {
+	string send_request(1:binary img_query),
+	void ping();
```
Comment (jhauswald) [inline imm/sirius/service.thrift]: does ping have a point?

File: imm/template/Makefile [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+					-lboost_thread \
+					-lboost_regex \
+					-lprotobuf \
+					-ltesseract \
+					-lthrift
+
+TARGET_CLIENT = imclient 
+TARGET_SERVER = imserver
+
+CMDOBJECT_SERVER = $(HOME)/sirius/sirius-application/command-center/gen-cpp/CommandCenter.o \
```
Comment (jhauswald) [inline imm/template/Makefile]: shouldn't be in the template Makefile

File: imm/sirius/service.thrift (hunk 2 above)
Comment (twurman) [inline imm/sirius/service.thrift]: its the heartbeat method for the command center to keep track of whether the service is alive or not

**Pull request closed without merging**